packages/router/router.ts

```diff
@@ -472,14 +477,18 @@ export function createRouter(init: RouterInit): Router {
     );
   }
 
+  let initialized =
+    !initialMatches.some((m) => m.route.loader) ||
+    (init.hydrationData != null && !foundMissingHydrationData);
+
   let router: Router;
   let state: RouterState = {
     historyAction: init.history.action,
     location: init.history.location,
     // If we do not match a user-provided-route, fall back to the root
     // to allow the errorElement to take over
     matches: initialMatches,
-    initialized: init.hydrationData != null && !foundMissingHydrationData,
```

Comment:
Noticed one small bug in this when I added the basename test (which used routes without loaders).  We should be immediately initialized if none of our matches have loaders